tests/net/bridge/net.eth_bridge failure on eth_mcux

**Describe the bug**
Build of tests/net/bridge/net.eth_bridge fails with:

/__w/zephyr/zephyr/drivers/ethernet/eth_mcux.c:1196: undefined reference to `net_if_mcast_mon_register'

Bisect identifies: 22d470e6a5c50513ac184a8ba84d8c62a279fd3d as the culprit

Inspection shows that the problem commit did not properly wrap code in the eth_mcux driver code with IPV4 | IPV6 condition so the result is an unconditional reference to a function in the IP stack that doesn't exist because the test case doesn't include the IP stack.

Unfortunately, CI for the commit apparently did not build this test.

Reproduce:

west build -b vmu_rt1170 -p always tests\net\bridge -T net.eth_bridge

--- a/drivers/ethernet/eth_mcux.c
+++ b/drivers/ethernet/eth_mcux.c
@@ -1163,6 +1163,7 @@ static int eth_init(const struct device *dev)
 	return 0;
 }
 
+#if defined(CONFIG_NET_NATIVE_IPV4) || defined(CONFIG_NET_NATIVE_IPV6)
 static void net_if_mcast_cb(struct net_if *iface,
 			    const struct net_addr *addr,
 			    bool is_joined)
@@ -1185,15 +1186,18 @@ static void net_if_mcast_cb(struct net_if *iface,
 		ENET_LeaveMulticastGroup(context->base, mac_addr.addr);
 	}
 }
+#endif /* CONFIG_NET_NATIVE_IPV4 || CONFIG_NET_NATIVE_IPV6 */
 
 static void eth_iface_init(struct net_if *iface)
 {
 	const struct device *dev = net_if_get_device(iface);
 	struct eth_context *context = dev->data;
 
+#if defined(CONFIG_NET_NATIVE_IPV4) || defined(CONFIG_NET_NATIVE_IPV6)
 	static struct net_if_mcast_monitor mon;
 
 	net_if_mcast_mon_register(&mon, iface, net_if_mcast_cb);
+#endif /* CONFIG_NET_NATIVE_IPV4 || CONFIG_NET_NATIVE_IPV6 */
 
 	net_if_set_link_addr(iface, context->mac_addr,
 			     sizeof(context->mac_addr),

